tgui/packages/tgui/interfaces/TurretControl.js

```diff
@@ -0,0 +1,108 @@
+import { Window } from '../layouts';
+import { useBackend, useLocalState } from '../backend';
+import { Button, Section, Flex, Box, Stack } from '../components';
+
+const randInt = (a, b) => {
+  if (a > b) {
+    let temp = a;
+    a = b;
+    b = temp;
+  }
+  return Math.floor(Math.random() * (b - a + 1)) + a;
+};
+
+// this is a totally real and normal semicolon I promise
+const glitches = ['$', '{', ']', '%', '^', '?', '>', '¬', 'π', ';', 'и', 'ю', '/', '#', '~'];
+const glitch = (text, amount) => {
+  for (let i = 0; i < amount; i++) {
+    let index = randInt(0, text.length);
+    text = text.slice(0, index) + glitches[randInt(0, glitches.length - 1)] + text.slice(index, text.length);
+  }
+  return text;
+};
+
+const generate_kill = (number) => {
+  let out = [];
+  for (let i = 0; i < 7; i++) {
+    if (Math.random() > 0.3) {
+      out.push("Kill. ");
+    } else {
+      out.push("KILL. ");
+    }
+  }
+  return out.map((kill, index) => (<Box inline preserveWhitespace fontSize={randInt(11, 25) + "px"} key={index}>{kill}</Box>));
+};
+
+export const TurretControl = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    enabled,
+    lethal,
+    emagged,
+    area,
+    locked,
+  } = data;
+
+  const set_lethal = (value) => {
+    if (value === !!lethal) {
+      return;
+    }
+    act("setLethal", { "lethal": value });
+  };
+  const set_enabled = (value) => {
+    if (value === !!enabled) {
+      return;
+    }
+    act("setEnabled", { "enabled": value });
+  };
+  return (
+    <Window
+      title={emagged ? "FATAL ERROR" : "Turret control (" + area + ")"}
```

Comment:
```suggestion
      title={emagged ? "FATAL ERROR" : `Turret control (${area})`}
```
Template literals make interpolation a little neater.